### ISSUE TITLE
Two tests failing locally for windows x64 checked

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/coreclr#8683]
I am seeing two tests fail with windows x64 checked builds. This is @8395eb5 (which is my change, but seemingly unrelated). Not sure why we don't see these failing in the lab.
```
JIT\Directed\nullabletypes\castclassvaluetype_r
JIT\Directed\nullabletypes\castclassvaluetype_d
```
Typical failure is:
```
Assert failure(PID 43884 [0x0000ab6c], Thread: 59336 [0xe7c8]): (uImageBase == uImageBaseFromOS) && (pFunctionEntry == pFunctionEntryFromOS)

CORECLR! Thread::VirtualUnwindCallFrame + 0x49A (0x00007ffa`71bf4306)
CORECLR! EECodeManager::EnsureCallerContextIsValid + 0x1F6 (0x00007ffa`71eaa2a2)
CORECLR! StackFrameIterator::CheckForSkippedFrames + 0x70 (0x00007ffa`71beea00)
CORECLR! StackFrameIterator::ProcessCurrentFrame + 0x160 (0x00007ffa`71bf3470)
CORECLR! StackFrameIterator::NextRaw + 0xE02 (0x00007ffa`71bf2dda)
CORECLR! StackFrameIterator::Next + 0xC1 (0x00007ffa`71bf1f19)
CORECLR! Thread::StackWalkFramesEx + 0x168 (0x00007ffa`71bf3ac8)
CORECLR! ExceptionTracker::FindNonvolatileRegisterPointers + 0x15B (0x00007ffa`71c0a61f)
CORECLR! FixNonvolatileRegisters + 0x155 (0x00007ffa`71c0c9d9)
CORECLR! ProcessCLRException + 0x8DE (0x00007ffa`71c18b2e)
    File: c:\repos\coreclr2\src\vm\stackwalk.cpp Line: 631
    Image: C:\repos\coreclr2\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

**Comment by AndyAyersMS** [comment]
This does not repro on a different local machine. The failing machine is newer so tried disabling AVX and SSE3_4 codegen but it did not change things any. Am going to do a clean rebuild on the "bad" machine next.

**Comment by AndyAyersMS** [comment]
Does not repro after a clean rebuild, so will close this...

**Comment by jkotas** [comment]
I am hitting with MusicStore on debug builds. 

1. Download https://dotnetcli.blob.core.windows.net/dotnet/preview/Binaries/1.0.0-preview2-1-003180/dotnet-dev-win-x64.1.0.0-preview2-1-003180.zip and extract it somewhere. I have extracted it to \dotnet-003180 so that's why I am going to use in the next steps.

2. Download https://dotnetcli.blob.core.windows.net/dotnet/master/Binaries/1.2.0-beta-001269-00/dotnet-win-x64.1.2.0-beta-001269-00.zip and extract it to same location (\dotnet-003180), overwriting the 3 files that are the same. If you see dotnet run complaining about missing runtime in step 3, update 001269 to whatever version it is complaining about.

3. Build debug build of CoreCLR and copy it over 

`copy \coreclr\bin\Product\Windows_NT.x64.Release \dotnet-003180\shared\Microsoft.NETCore.App\1.2.0-beta-001269-00`

4. Clone, build and run music store:

git clone https://github.com/aspnet/MusicStore
cd MusicStore/samples/MusicStore
\dotnet-003180\dotnet restore
\dotnet-003180\dotnet run

Result: The above assert

**Comment by rahku** [comment]
I do not hit the above assert with the latest code. But I do hit another assert at https://github.com/dotnet/coreclr/blob/master/src/vm/assemblyspec.cpp#L1509
F:\gist\musicstore\MusicStore\samples\MusicStore>"C:\Users\rahku\Downloads\dotnet-win-x64.latest\dotnet.exe" run --framework netcoreapp1.1

Assert failure(PID 36920 [0x00009038], Thread: 13848 [0x3618]): Postcondition failure: FAILED: UnsafeContains(this, pSpec)

**Comment by rahku** [comment]
And I have opened PR to fix the above assert

**Comment by briansull** [comment]
Note that the assert was fixed with PR #10116
